components/verifalia/actions/verify-email/verify-email.mjs

```diff
@@ -0,0 +1,189 @@
+      // the one we are allowed by Pipedream, we just let the user know.
+
+      let resumeUrl = null;
+      let waitForResults = false;
+      let cancellationToken = null;
+
+      if ($.context.test) {
+        // In test mode, we will wait for the verification results during this execution
+        // and cancel the token after 20s (the actual lowest limit is 30s).
+        // See: https://pipedream.com/docs/limits/#time-per-execution
+
+        waitForResults = true;
+        cancellationToken = new CancellationToken();
+        setTimeout(() => cancellationToken.cancel(), 20 * 1000);
+      } else {
+        // In deployment mode, we won't wait for the verification results during this
+        // execution and, instead, rely on the Verifalia completion callback support and
+        // on the Pipedream's new ability to suspend execution flows.
+
+        resumeUrl = $.flow.rerun(86400 * 1000, {}, 1).resume_url;
+      }
+
+      // Submit the verification to Verifalia
+
+      let job;
+
+      try {
+        job = await this.verifalia.wrapVerifaliaApiInvocation(async () => {
+          return await verifaliaClient
+            .emailValidations
+            .submit({
+              quality: this.quality,
+              entries: [
+                {
+                  inputData: this.emailAddress,
+                },
+              ],
+              retention: this.retention,
+              callback: {
+                url: resumeUrl,
+              },
+            },
+            waitForResults,
+            cancellationToken);
+        });
+      } catch (error) {
+        // If the error is an OperationCanceledError it means that we are in test mode and
+        // have manually cancelled the submission: just let the user know why that happened.
+
+        if (error instanceof OperationCanceledError) {
+          // Had to replace the word req*ire with a synonym because of https://github.com/PipedreamHQ/pipedream/issues/3187 :)
+
+          throw new Error("This operation would need some more time to complete and that would not work properly in " +
+                        "Pipedream's test mode. Please deploy your workflow to get a meaningful email verification result.");
+        }
+
+        throw error;
+      }
+
+      // Handle the verification outcome, if available
+
+      if (job.overview.status === "Completed") {
+        if (!resumeUrl) {
+          // We are in test mode and the validation just completed in a single request
+
+          return job;
+        }
+
+        // TODO: If the validation completed, abort the suspension and return the results
+        // Blocked by https://github.com/PipedreamHQ/pipedream/issues/3094
+
+        // We are *not* in test mode and the job is complete: simulate an instantaneous
+        // webhook completion callback, as the external one may take a few seconds to
+        // fire and we wish to continue the flow as fast as possible.
+
+        return await axios({
```

Comment:
Could you please move this request to `app` and call it here?

Reply:
I believe this is somewhat overkill and I would prefer to leave it as is; also, that axios request is a temporary hack until https://github.com/PipedreamHQ/pipedream/issues/3151 will be fixed / closed. @vellames-turing could you proceed with the merge anyway, please?